### ISSUE TITLE
BUG- Event spanning all day are on day too short.

### ISSUE THREAD

**Issue by psnma** [fullcalendar/fullcalendar#270]
I'm passing to fullcalendar an event like this one: ```{id: 31, title: 'Test', start: '2014-12-22', end: '2014-12-23', allDay: true }``` I expected to find in the calendar an event that spans two whole days, but the event is only in the 2014-12-22 slot, not in the 2014-12-23 one.

**Comment by jamesvanleuven** [comment]
if you don't
mind me asking - did you try setting datetime with time as 00:00:00 to 23:59:59? 

James Mendham
Founder

"be without fear in the face of your enemies. 
Be brave and upright that God may love thee. 
Speak the truth always even if it leads to your death. 
Safeguard the helpless and do no wrong"



james@bookt.in  | +1 (604) 721-5474

CONFIDENTIAL:This electronic mail (including any attachments) may contain information that is privileged, confidential, and/or otherwise protected from disclosure to anyone other than its intended recipient(s). Any dissemination or use of this electronic email or its contents (including any attachments) by persons other than the intended recipient(s) is strictly prohibited. If you have received this message in error, please notify me immediately by reply email so that I may correct my records. Please then delete the original message (including any attachments) in its entirety. Thank you

**Comment by psnma** [comment]
I am passing a date, not a datetime, just ```2014-12-22``` until ```2014-12-23```. It's a ````allDay = true```.
Need to fix that bug.

**Comment by arshaw** [comment]
this is because all-day end dates are exclusive. see the `end` property in this article http://fullcalendar.io/docs/event_data/Event_Object/

**Comment by damangio** [comment]
I understand that the intended behavior is to treat all end dates as exclusive, but I think this doesn't make sense for all-day events, especially from the perspective of a user.

If a user creates an all-day event beginning on the 10th and ending on the 12th, they will expect it to show up on the 10th, 11th, and 12th, not just the 10th and 11th. In order to make my application behave intuitively, I have had to adjust my end dates of all-day events by 1 day whenever they are loaded into FullCalendar, and adjust them back whenever they are retrieved from FullCalendar.

I'm not saying it isn't a good design to treat end dates as exclusive, but I think it is a bad assumption to translate plain dates into midnight. That does not jive with the concept of a date. A date is more abstract than a timestamp. When a user says, "This event ends on May 12, 2016," they do not mean that it ends at 2016-05-12 00:00:00.

In summary, I'm suggesting that all-day events should have different logical treatment.

If there is a better place for discussing this, let me know. Thanks.

**Comment by arshaw** [comment]
that's actually what fullcalendar used to do in v1.3.0-v1.6.4, but that was even MORE confusing to users because it conflicted with the way timed events' end times worked.

FWIW, iCal, a pretty standard event format, does the same thing
